js/_website/generate_jsons/src/docs/__init__.py

```diff
@@ -54,35 +54,38 @@ def add_demos():
 
 add_demos()
 
-ordered_events = [
-    "Change()",
-    "Click()",
-    "Submit()",
-    "Edit()",
-    "Clear()",
-    "Play()",
-    "Pause()",
-    "Stream()",
-    "Blur()",
-    "Upload()",
-]
-
-
-def add_supported_events():
+events = [
```

Comment:
How can we keep this up to date in the future?

Maybe we can programmatically import all `EventListeners` and programmatically document all the properties that are of instance `EventListenerMethod`. Or we have `EventListenerMethod` update a global dict with the event name?

Reply:
so I actually got stuck on this earlier: when I list the attributes of one of the EventListeners classes, it doesn't show any of the EventListenerMethods.. not sure why.

```
<class 'gradio.events.Changeable'>
{'__module__': 'gradio.events', '__init__': <function Changeable.__init__ at 0x7fcf3004f1f0>, '__doc__': None}
```

Reply:
just fixed it @freddyaboulton, not hardcoded anymore. can you take another look? 

Btw what is 'end'? not showing up as a method for any component

Reply:
The end event should be triggered when the video/audio stops playing. We should add it to document decorator for Playable class. I think that was an oversight.

Reply:
got it thanks! added it, and showing up correctly with video and audio :) will merge. thanks for the help!